conda_lock/pypi_solver.py

```diff
@@ -0,0 +1,286 @@
+import re
+import sys
+
+from pathlib import Path
+from typing import Dict, List, Optional, TypedDict
+from urllib.parse import urldefrag
+
+from clikit.api.io.flags import VERY_VERBOSE
+from clikit.io import ConsoleIO, NullIO
+from packaging.tags import compatible_tags, cpython_tags
+from poetry.core.packages import Dependency, Package, ProjectPackage, URLDependency
+from poetry.installation.chooser import Chooser
+from poetry.installation.operations import Install
+from poetry.installation.operations.uninstall import Uninstall
+from poetry.puzzle import Solver
+from poetry.repositories.pool import Pool
+from poetry.repositories.pypi_repository import PyPiRepository
+from poetry.repositories.repository import Repository
+from poetry.utils.env import Env
+
+from conda_lock import src_parser
+from conda_lock.src_parser.pyproject_toml import get_lookup as get_forward_lookup
+from conda_lock.src_parser.pyproject_toml import normalize_pypi_name
+
+
+class PlatformEnv(Env):
+    def __init__(self, python_version, platform):
+        super().__init__(path=Path(sys.prefix))
+        if platform == "linux-64":
+            # FIXME: in principle these depend on the glibc in the conda env
+            self._platforms = ["manylinux_2_17_x86_64", "manylinux2014_x86_64"]
```

Comment:
should we toss `manylinux1`  in there?

Reply:
Should probably add in `manylinux_2_17_aarch64` at least for the arm stuff.  Used in eg numpy https://pypi.org/project/numpy/#files

Reply:
Okay, I added support for PEP 513, 599, and 600 tags.